internal/dao/sqlx/source_dao.go

```diff
@@ -0,0 +1,94 @@
+package sqlx
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/RHEnVision/provisioning-backend/internal/db"
+	m "github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/jmoiron/sqlx"
+)
+
+const (
+	create     = `INSERT INTO sources (source_id, name, org_id, auth_id) VALUES ($1, $2, $3, $4) RETURNING id,created_at`
+	getById    = `SELECT * FROM sources WHERE source_id = $1;`
+	list       = `SELECT * FROM sources ORDER BY id LIMIT $1 OFFSET $2`
+	getByOrgId = `SELECT * FROM sources WHERE org_id = $1 ORDER BY id LIMIT $2;`
+)
+
+type sourceDaoSqlx struct {
+	getById    *sqlx.Stmt
+	list       *sqlx.Stmt
+	create     *sqlx.Stmt
+	getByOrgId *sqlx.Stmt
+}
+
+var GetSourceDao func(ctx context.Context) (dao.SourceDao, error)
+
+func getSourceDao(ctx context.Context) (dao.SourceDao, error) {
+	var err error
+	daoImpl := sourceDaoSqlx{}
+
+	daoImpl.getById, err = db.DB.PreparexContext(ctx, getById)
+	if err != nil {
+		fmt.Println(err.Error())
+		return nil, err
+	}
+	daoImpl.list, err = db.DB.PreparexContext(ctx, list)
+	if err != nil {
+		fmt.Println(err.Error())
+		return nil, err
+	}
+	daoImpl.create, err = db.DB.PreparexContext(ctx, create)
+	if err != nil {
+		fmt.Println(err.Error())
+		return nil, err
+	}
+	daoImpl.getByOrgId, err = db.DB.PreparexContext(ctx, getByOrgId)
+	if err != nil {
+		fmt.Println(err.Error())
+		return nil, err
+	}
+	return &daoImpl, nil
+}
+
+func init() {
+	GetSourceDao = getSourceDao
+}
+
+func (s *sourceDaoSqlx) Create(source *m.Source) error {
+	_, err := db.DB.Exec("INSERT INTO sources (source_id, name, org_id, auth_id) VALUES ($1, $2, $3, $4) RETURNING created_at", source.SourceID, source.Name, source.OrgID, source.AuthID)
+	//_, err := s.create.Exec(source)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *sourceDaoSqlx) GetById(id uint64) (*m.Source, error) {
+	result := &m.Source{}
+	err := s.getById.Get(result, id)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func (s *sourceDaoSqlx) List(limit uint64, offset uint64) ([]*m.Source, error) {
+	result := []*m.Source{}
+	err := s.list.Select(&result, limit, offset)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func (s *sourceDaoSqlx) GetByOrgId(orgId uint64, limit uint64) ([]*m.Source, error) {
+	result := []*m.Source{}
+	err := s.getByOrgId.Select(&result, orgId, limit)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
```

Comment:
I have a problem here, maybe you would know why it happens. 
I am running: 
```
var orgID uint64 = 1
var limit uint64 = 1
sourcesArr, err := sourceDao.GetByOrgId(orgID, limit)
```
and I am getting : 
`error="cannot convert 1 to Text"
`
Why is that a problem? I saw you did the same.